modules/gapi/include/opencv2/gapi/pattern_matching.hpp

```diff
@@ -0,0 +1,52 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2018 Intel Corporation
+
+
+#ifndef OPENCV_GAPI_PATTERN_MATCHING_HPP
+#define OPENCV_GAPI_PATTERN_MATCHING_HPP
+
+#include <functional> 
+#include <unordered_set>
+#include <unordered_map>
+#include <list>
+
+#include "opencv2/gapi/gcomputation.hpp"
+#include "opencv2/gapi/gcompiled.hpp"
+#include "opencv2/gapi/gkernel.hpp"
+
+#include "api/gcomputation_priv.hpp"
+#include "api/gcall_priv.hpp"
+#include "api/gnode_priv.hpp"
+
+#include "compiler/gcompiled_priv.hpp"
+#include "compiler/gmodel.hpp"
+
+#include <ade/graph.hpp>
+#include <ade/typed_graph.hpp>
```

Comment:
I believe this include list can be reduced (many headers can be placed in .cpp file)

Reply:
Done